api_handlers.go

```diff
@@ -257,3 +257,52 @@ func getLastCompleteRunSHA(ctx context.Context, browserNames []string) (sha stri
 	}
 	return "latest", nil
 }
+
+// apiDiffHandler takes 2 test-run results JSON files and produces
+// JSON in the same format, with only the differences in runs.
+func apiDiffHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := appengine.NewContext(r)
+
+	var err error
+	params, err := url.ParseQuery(r.URL.RawQuery)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	specBefore := params.Get("before")
+	if specBefore == "" {
+		http.Error(w, "before param missing", http.StatusBadRequest)
+		return
+	}
+	var beforeJSON map[string][]int
+	if beforeJSON, err = fetchRunResultsJSONForParam(ctx, r, specBefore); err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	} else if beforeJSON == nil {
+		http.Error(w, specBefore+" not found", http.StatusNotFound)
+		return
+	}
+
+	specAfter := params.Get("after")
+	if specAfter == "" {
+		http.Error(w, "after param missing", http.StatusBadRequest)
+		return
+	}
+	var afterJSON map[string][]int
+	if afterJSON, err = fetchRunResultsJSONForParam(ctx, r, specAfter); err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	} else if afterJSON == nil {
+		http.Error(w, specAfter+" not found", http.StatusNotFound)
+		return
+	}
+
+	diffJSON := diffResults(beforeJSON, afterJSON)
```

Comment:
My first time reading go code. If I'm reading this right, almost everything up to this point is error handling :)

Reply:
Yeah, go's error handling is .. verbose.